### PR TITLE
Pythonise Memory Allocation (memory_allocation.rst)

### PR DESCRIPTION
**Pull request by 0dminnimda**
Continuing to Pythonize Сython documentation!

This PR Pythonizes [this](https://cython.readthedocs.io/en/latest/src/tutorial/memory_allocation.html) documentation page.

Lots of process details and standardization are found here: #4187

### CODE REVIEW

**Comment by 0dminnimda** [comment]
@scoder this PR is ready for review, the only thing that you might not like is that I have expanded some function calls to two lines for ease of reading but if it's fine, then this PR is ready for merge.

File: docs/examples/tutorial/memory_allocation/some_memory.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from cython.cimports.cpython.mem import PyMem_Malloc, PyMem_Realloc, PyMem_Free
+
+@cython.cclass
+class SomeMemory:
+    data: cython.p_double
+
+    def __cinit__(self, number: size_t):
```
Comment (matusvalo) [inline docs/examples/tutorial/memory_allocation/some_memory.py:7]:
```suggestion
    def __cinit__(self, number: cython.size_t):
```

File: docs/examples/tutorial/memory_allocation/some_memory.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        # allocate some memory (uninitialised, may contain arbitrary data)
+        self.data = cython.cast(cython.p_double, PyMem_Malloc(
+            number * cython.sizeof(cython.double)))
+        if not self.data:
+            raise MemoryError()
+
+    def resize(self, new_number: size_t):
```
Comment (matusvalo) [inline docs/examples/tutorial/memory_allocation/some_memory.py:14]:
```suggestion
    def resize(self, new_number: cython.size_t):
```

**Review by matusvalo — COMMENTED** [review]
Except two comments LGTM

**Comment by 0dminnimda** [comment]
> Except two comments LGTM

Good point, I overlooked this, which is interesting, it successfully cythonizes

**Comment by 0dminnimda** [comment]
something breaks with py2 due to using `cython.size_t` instead of `size_t`, hhmm

**Comment by matusvalo** [comment]
> something breaks with py2 due to using cython.size_t instead of size_t, hhmm

Could you describe what is breaking? I was able to compile the code correctly.

**Comment by 0dminnimda** [comment]
> > something breaks with py2 due to using cython.size_t instead of size_t, hhmm
> 
> Could you describe what is breaking? I was able to compile the code correctly.

look at the CI output ...

**Comment by 0dminnimda** [comment]
`UnicodeEncodeError: 'ascii' codec can't encode characters in position X-Y: ordinal not in range(128)`

**Comment by 0dminnimda** [comment]
@matusvalo this is probably a github action issue for this repo, all recent tests show the same problem but on the fork everything is fine

**Comment by 0dminnimda** [comment]
open and close to rerun ci

**Comment by 0dminnimda** [comment]
@scoder, if you don't have any comments, then this PR is ready to be merged

**Pull request merged**